streaming/java/streaming-api/src/main/java/org/ray/streaming/api/context/StreamRuntimeContext.java

```diff
@@ -3,7 +3,7 @@
 /**
  * Encapsulate the runtime information of a streaming task.
  */
-public interface RuntimeContext {
+public interface StreamRuntimeContext {
```

Comment:
Will it better to name ti as `StreamingRuntimeContext`? It's also consist with `StreamingContext`.

Reply:
Yes that would be a point. I'd prefer unify all appearance meaning "Streaming Processing" v.s. "Batch Processing" as "Streaming". Will update if there is no objection.

Reply:
This is one place that flink didn't handle very clean. Please take a look at the flink code. They are using both "streaming" and "stream" with no clear rule. BTW, if you google a little bit, you will find that there is no "Streaming processing". Most people are using "stream processing", i.e. wikipedia or even ververica(https://www.ververica.com/what-is-stream-processing). "Streaming" as the present continuous verb tense is used to emphasize the "doing" status. So, IMHO, the StreamingContext should be changed to StreamContext.

Reply:
@JingGe I agree that in Flink there are both appearances "Streaming" and "Stream" which can be unified. Thanks for your information. I will check the source you provide. If we then rename "StreamingContext", it can be in a separated pr.

Reply:
I will create another pr for refactoring the StreamingContext into StreamContext.

Reply:
I prefer `StreamingContext`. It's more intuitional to me as entry point of ray streaming

Reply:
`Streaming` itself refers to `Stream processing`?

Reply:
So `StreamingContext` has a meaning of `StreamProcessingContext`?

Reply:
Streaming and StreamProcessing have obviously different meanings. It is quite difficult to get the info of "Processing" from "Streaming".

Reply:
There are also other usage of "Stream" in the source code like StreamSink, StreamSource, StreamOperator, UnionStream, and many many more.

Reply:
It's different. Using `Stream` in StreamSink, StreamSource, StreamOperator, UnionStream is because they all are data stream. It need to be a noun. In `StreamingContext` I think a verb is more intuitional.

Reply:
First of all, “streaming” is not a verb, "stream" is a verb. Second, if we have used the "stream" for all data stream, why not use it for the context? StreamSink, StreamSource, StreamOperator, UnionStream mean the sink, source, operator, and union of the stream, and StreamContext means the context of the stream. It is consistent.

Reply:
I notice that flink uses `StreamingRuntimeContext` in operator processing.
So It's about expressing a dynamic relationship. Specially, StreamingRuntimeContext may contain state that's transient.

Reply:
Please take a look at StreamContextEnvironment, StreamOptimizeContext, etc. in Flink. Like I mentioned before, this is a place where Flink did take care of.

Reply:
I just checked `StreamOptimizeContext`, seems it's used to optimize execution plan which is more related to data stream. So using `Stream` is appropriate there. And for `StreamingContext`, flink doesn't has this class, it doesn't come from flink.

Reply:
It's not that we can't reuse this name  if other system else uses it. It's a consensus, so we do not need to deliberately mark the new innovative terms. In my submission , `StreamingRuntimeContext` is better than `StreamRuntimeContext` if need to add some prefix.

Reply:
If all systems using `streaming` and there is no strong reason to replace it, we should keep that name. So user can know what it means directly, otherwise he might wonder why it's not called `streaming`. And if we replace all `streaming` to `stream`, will we replace `ray streaming` to `ray stream`? I think it's more natural to associate `ray streaming` with `StreamingContext` instead of `StreamContext`.

Reply:
One additional suggestion: it might be a good idea that, for those internal used classes, we'd better avoid using the exactly same class name as Flink used for the same purpose.

> I just checked `StreamOptimizeContext`, seems it's used to optimize execution plan which is more related to data stream. So using `Stream` is appropriate there. And for `StreamingContext`, flink doesn't has this class, it doesn't come from flink.

Both StreamOptimizeContext and StreamRuntimeContext are contexts. Why StreamOptimizeContext is fine to you and StreamRuntimeContext is not? It is quite confused.

Reply:
> If all systems using `streaming` and there is no strong reason to replace it, we should keep that name. So user can know what it means directly, otherwise he might wonder why it's not called `streaming`. And if we replace all `streaming` to `stream`, will we replace `ray streaming` to `ray stream`? I think it's more natural to associate `ray streaming` with `StreamingContext` instead of `StreamContext`.

Please don't mix the module or engine name with the java class name. I have no problem at all to have engine named ray streaming. BTW, what are the "all systems" who are using streaming?

Reply:
`StreamOptimizeContext ` is used for optimize data stream pipeline. But `StreamRuntimeContext ` has nothing with data stream.

Reply:
> `StreamOptimizeContext ` is used for optimize data stream pipeline. But `StreamRuntimeContext ` has nothing with data stream.

To be honest, data stream is the fundation of the whole stream processing. Every context must have something to do with the data stream. It is interesting to know that you think StreamRuntimeContext has nothing to do with the data stream. If this is true, why should such context exist in the first place.

Let get to the root. What is the strong different you feel between StreamRuntimeContext and StreamingRuntimeContext? I have no problem to use either of them. But if we were talking about the clean consist naming convention, I would prefer StreamRuntimeContext. Just my 2cents, feel free to take it or leave it. Thanks for the interesting discussion.

Reply:
ALL SYSTEMS might mean we can observe Spark uses `DataStream` in operators and `StreamingContext` in processing and Flink is borrowing same concepts from Spark (named Data Stream and StreamingRuntimeContext).

Reply:
`StreamRuntimeContext`/`StreamingRuntimeContext ` is a runtime context for user function execution. The user can get something like task id/index/parallelism about the task that executes the user function.

Reply:
`RuntimeContext` here is mainly from the view of the api user. The information it contains is mainly for a parallel sub task which execute the user function. Maybe we can name it as `TaskContext` or something else.

Reply:
As far as `streaming` or `stream`, either of them is ok to me if the user feels it's ok

Reply:
Again, interesting discussion, one last try. I'd like to share the different between "stream" and "streaming" from the word meaning perspective so far I understood. "Streaming" is a kind of macro virtual concept, it means everything about the stream processing, that is why it is fine to call the module ray streaming at the concept level. There is no "a" streaming, and we cannot define the runtime context of "a" streaming. On the other hand, "stream" is a concrete object, each DAG we build (based on data streams, stream operators, etc.), deploy, and run, is "a" stream. It is nature to say each running stream has a runtime context, therefore we got the name StreamRuntimeContext which means the runtime context of a stream.

Reply:
From this point of view, it makes sense to me. Please change name in python streaming too.

Reply:
Thanks for your discussion so far. It seems reasonable to me that for the meaning a context of a <placeholder> the placeholder is stream, which refers to a concrete object.

Possibly we have a wiki page or somewhere documented glossary(and the discussion) so that later one can easily refer to.